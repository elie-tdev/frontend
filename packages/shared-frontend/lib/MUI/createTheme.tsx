import { createTheme as createMuiTheme } from '@mui/material/styles'

import { makeDefaultTheme } from './defaultTheme'
import { BAIThemeOptions } from './types'
import { createDefaultPaletteOptions } from './utils/palette'
import { createDefaultSpacingOptions } from './utils/spacing'

const createThemeBase = (themeOptions: BAIThemeOptions, isTest = false) => {
  createDefaultSpacingOptions(themeOptions)
  createDefaultPaletteOptions(themeOptions)
  return makeDefaultTheme(createMuiTheme(themeOptions), !isTest)
}

export const createTheme = (themeOptions: BAIThemeOptions = {}) =>
  createThemeBase(themeOptions)

export const createThemeTest = (themeOptions: BAIThemeOptions = {}) =>
  createThemeBase(themeOptions, true)
