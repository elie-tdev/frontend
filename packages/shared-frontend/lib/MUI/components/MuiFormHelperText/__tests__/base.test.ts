import { MuiFormHelperText } from '..'
import { createThemeTest } from '../../../createTheme'

describe('FormHelperText', () => {
  const theme = createThemeTest()
  const component: any = MuiFormHelperText(theme)

  it('should change default element type to "div"', () => {
    expect(component?.defaultProps?.component).toBe('div')
  })

  it('should be consistent with previous versions: ', () => {
    expect(component).toMatchSnapshot()
  })
})
