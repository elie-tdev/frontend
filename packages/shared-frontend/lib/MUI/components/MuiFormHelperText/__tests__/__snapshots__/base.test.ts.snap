// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormHelperText should be consistent with previous versions:  1`] = `
Object {
  "defaultProps": Object {
    "component": "div",
  },
}
`;
